src/main/java/executor/command/CommandAddSpendingReceipt.java

```diff
@@ -0,0 +1,102 @@
+package executor.command;
+
+import executor.task.TaskList;
+import interpreter.Parser;
+import ui.Receipt;
+import ui.Ui;
+import ui.Wallet;
+
+import java.util.ArrayList;
+import java.util.Date;
+
+public class CommandAddSpendingReceipt extends Command {
+
+    private String userInput;
+    private Double spending;
+    private Date date;
+    private ArrayList<String> tags;
+    private String description;
+
+    /**
+     * Constructor for CommandAddSpendingReceipt subCommand Class.
+     * @param userInput The user input from the CLI
+     */
+    public CommandAddSpendingReceipt(String userInput) {
+        this.commandType = CommandType.OUT;
+        this.userInput = userInput;
+        this.description = "You can add a new spendings receipt in format of 'Out $5.00 /tags tag'.";;
+
+    }
+
+    @Override
+    public void execute(TaskList taskList) {
+    }
+
+    @Override
+    public void execute(Wallet wallet) {
+        parseUserInput();
+        Receipt r = new Receipt(this.spending, this.date, this.tags);
+    }
+
+    private String parseForTags(String temp) {
```

Comment:
A lot of functions are redundant it seems. Why not move these to a parent class to reduce redundancy?